code-challenges.test.js

```diff
@@ -11,37 +14,156 @@
 // Run the file with the following command: $ yarn jest
 
 
+
+
+
+
+
+
+
+
+
+
 // --------------------1) Create a function that takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence.
 
-// a) Create a test with expect statements for each of the variables provided.
 
 const fibLength1 = 6
 // Expected output: [1, 1, 2, 3, 5, 8]
 
 const fibLength2 = 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+// a) Create a test with expect statements for each of the variables provided.
+
+describe ("fibonacci", () => {
+    it ("Takes in a number and returns the the fibonacci sequence, its length is dependent on the input number value ", () => {
+        expect ( fibonacci(fibLength1) ).toEqual( [1, 1, 2, 3, 5, 8] )
+        expect ( fibonacci(fibLength2) ).toEqual( [1, 1, 2, 3, 5, 8, 13, 21, 34, 55] )
+    })
+})
+
+// Good Failure:
+// FAIL  ./code-challenges.test.js
+//   fibonacci
+//     ✕ Takes in a number and returns the the fibonacci sequence, its length is dependent on the input number value  (1 ms)
+//   ● fibonacci › Takes in a number and returns the the fibonacci sequence, its length is dependent on the input number value 
+//     ReferenceError: fibonacci is not defined
+
+
+
+
+
 
 // b) Create the function that makes the test pass.
+// Psuedocode: 
+// Create a function called fibonacci
+// Input are numbers only--the only restraint is that the input numbers must be greater than 2 
+// Output is the fibonacci seqeuence based on the value inputed in the parameter and fitting the condtions
+// Logic in general ==> (a number @ index) + (a number @ index - 1) = (the number @ index + 1)
+// We are going to either have: 1) A container array that collects new numbers in different array(s)? or 2) A array that utilizes the .map() and produces a new array --since its an acessor-- for the user
+
+const fibonacci = (number) => {
+    // 2 arrays: fibonacciArray is out container array; starterFibonacci is only useful with the 0th - 1st index
+    let fibonacciArray = [1]
+    let starterFibonacci = [0, 1]
+for (let index = 0 ; index < number - 1 ; index = index + 1 ) {
+    // 1st index of fibonacciArray is 0 + 1 
+    if (index === 0 || index === 1){
+    fibonacciArray.push(starterFibonacci[index] + fibonacciArray[index])
+    } else if (index > 1){
+    // 2nd index and onwards of fibonacciArray; 1 + 1 = 2
+    fibonacciArray.push(fibonacciArray[index] + fibonacciArray[index - 1])
+    } else {
+    }
+}   return fibonacciArray
+}
+
+// Refactoring with map will make this code more cleaner, since .map() method produces a new array without recalling or invoking anything. + we can set 2 conditional statements using   .map( (value, index) => { ... } )
```

Comment:
Good idea! problem is though map only works on arrays of the right length so we can start off with an Array.fill(...)